### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by nini-faroux**
Adds tests

### CODE REVIEW

File: triphut-specs/Spec/AlwaysSucceed/Script.hs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  mkTypedValidator $$(PlutusTx.compile [||toBuiltinValidator succeedValidator1||])
+
+-- For upgrade minter testing
+type AlwaysSucceedUpgradeMinter = TypedPolicy ()
+
+alwaysSucceedTypedMintingPolicy :: AlwaysSucceedUpgradeMinter
+alwaysSucceedTypedMintingPolicy =
+  mkTypedPolicy $$(PlutusTx.compile [||toBuiltinPolicy succeedPolicy||])
+
+succeedPolicy :: BuiltinData -> ScriptContext -> Bool
```
Comment (Randy1Burrell) [inline triphut-specs/Spec/AlwaysSucceed/Script.hs:48]: Why have a succeedPolicy that takes two arguments, then throw away the params below?

File: triphut-specs/Spec/AlwaysSucceed/Script.hs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+succeedPolicy _ _ = True
```
Comment (Randy1Burrell) [inline triphut-specs/Spec/AlwaysSucceed/Script.hs:49]: What is the purpose of just returning `True` without doing anything?

**Review by Randy1Burrell — APPROVED** [review]
I left a comment. Nothing big, just a small question, but it wouldn't hold up the PR.

File: triphut-specs/Spec/AlwaysSucceed/Script.hs (hunk 2 above)
Comment (nini-faroux) [reply triphut-specs/Spec/AlwaysSucceed/Script.hs:49]: Oh yes, this is just an always succeed script used as a placeholder for the tests, so basically the way the protocol works is there an extra upgrade minting policy which some of the validation logic is deferred to, as mentioned in the readme, and so when attempting an upgrade the `validateConfiguration` script requires us to have used this specific upgrade policy to mint an upgrade token in this [check](https://github.com/yaadlabs/DAO/blob/a16f5f9cbdaab9ae71d7d09a623abbb4ac9e82db/triphut/Triphut/ConfigurationNft/Script.hs#L292).

And so in the tests I used this policy [here](https://github.com/yaadlabs/DAO/blob/a16f5f9cbdaab9ae71d7d09a623abbb4ac9e82db/triphut-specs/Spec/Upgrade/Context.hs#L149) in order to just have some placeholder policy.

It's something I wanted to talk to you about generally next week though.

**Comment by nini-faroux** [comment]
Thanks @Randy1Burrell , answered your question there, going to merge now.

**Pull request merged**